### DISCUSSION TITLE
A way to hint preferred positioning in the graph

### DISCUSSION THREAD

**Discussion by popovitsj** [reaviz/reagraph#117] [Q&A]
I'm using Reagraph to have a simplified way of displaying a world map. This is for a board game, where each country is adjacent to certain other countries.

Reagraph seems like a good fit, but where I'm stuck is that there doesn't seem to be a way to hint the layout of the graph. For example. I want Norway to be above of Germany, and Western Europe to be left of Germany.

It would be great if there was some way to hint the algorithm as to the relative position of nodes.

For example, in the image below, you can imagine how I'd like to play around with different hints to make the nodes displayed in their proper positions.

![image](https://github.com/reaviz/reagraph/assets/7157048/accf3f4d-f0e7-41a7-8184-77df355242a1)

Is there any way of doing this?

**Comment by popovitsj** [comment]
It would actually already help me a lot if there was some way to "save" a current layout, reached after dragging the nodes, so that this layout can be loaded next time.

**Reply by amcdnl** [reply]
that shouldnt be too difficult and might help with some other problems im exploring. ill take a stab at this this week

**Reply by popovitsj** [reply]
Do you have an idea on what the API for this would look like?

**Reply by amcdnl** [reply]
It would be a key/value pair of nodes + x/y/z position then a callback for position changes.

**Reply by popovitsj** [reply]
Yes, that's makes sense. So that would be for a method for saving/restoring a certain layout? I guess my initial question is a bit more vague, to have some "hinting" for the positioning.

**Reply by amcdnl** [reply]
The layout overrides provide a way to do this now.

**Comment by popovitsj** [comment]
Very nice, I saw you released the custom layout.
